multiversx_sdk_core/tokens.py

```diff
@@ -39,23 +39,33 @@ def extract_identifier_from_extended_identifier(self, identifier: str) -> str:
         parts = identifier.split("-")
 
         self._check_if_extended_identifier_was_provided(parts)
+        self._ensure_token_ticker_validity(parts[0])
         self._check_length_of_random_sequence(parts[1])
 
         return parts[0] + "-" + parts[1]
 
     def _check_if_extended_identifier_was_provided(self, token_parts: List[str]) -> None:
-        EXTENDED_IDENTIFIER_LENGTH_IF_SPLITTED = 3
+        # this is for the identifiers of fungible tokens
+        MIN_EXTENDED_IDENTIFIER_LENGTH_IF_SPLITTED = 2
+        # this is for the identifiers of nft, sft and meta-esdt
+        MAX_EXTENDED_IDENTIFIER_LENGTH_IF_SPLITTED = 3
 
-        if len(token_parts) != EXTENDED_IDENTIFIER_LENGTH_IF_SPLITTED:
+        if len(token_parts) < MIN_EXTENDED_IDENTIFIER_LENGTH_IF_SPLITTED or len(token_parts) > MAX_EXTENDED_IDENTIFIER_LENGTH_IF_SPLITTED:
             raise InvalidTokenIdentifierError("You have not provided the extended identifier")
```

Comment:
"Invalid extended token identifier provided" since all other errors are at the 3rd person